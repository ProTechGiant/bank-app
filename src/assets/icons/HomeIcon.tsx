import Svg, { G, Path } from "react-native-svg";

import type { IconProps } from "@/assets/icons";

export function HomeIcon({ width = 24, height = 24, color = "#666666" }: IconProps) {
  return (
    <Svg width={width} height={height} viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
      <G id="home">
        <Path id="Vector" d="M6 19H9V13H15V19H18V10L12 5.5L6 10V19ZM4 21V9L12 3L20 9V21H13V15H11V21H4Z" fill={color} />
      </G>
    </Svg>
  );
}
