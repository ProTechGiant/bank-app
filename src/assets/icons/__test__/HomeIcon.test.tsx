import { create } from "react-test-renderer";

import { HomeIcon } from "../HomeIcon";

describe("icon/HomeIcon", () => {
  it("renders a HomeIcon", () => {
    const result = create(<HomeIcon />);

    expect(result).toMatchSnapshot();
  });
});
