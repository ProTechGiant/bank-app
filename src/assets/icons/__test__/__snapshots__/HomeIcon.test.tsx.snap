// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon/HomeIcon renders a HomeIcon 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      name="home"
    >
      <RNSVGPath
        d="M6 19H9V13H15V19H18V10L12 5.5L6 10V19ZM4 21V9L12 3L20 9V21H13V15H11V21H4Z"
        fill={
          {
            "payload": 4284900966,
            "type": 0,
          }
        }
        name="Vector"
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGGroup>
</RNSVGSvgView>
`;
